runtime/doc/msgpack_rpc.txt

```diff
@@ -413,5 +417,30 @@ of update.
 ["popupmenu_hide"]
 	The popupmenu is hidden.
 
+["cmdline_enter"]
+	Enter the cmdline.
+
+["cmdline_leave"]
+	Leave the cmdline.
+
+["cmdline_firstc", firstc]
+        The first character of the command, which could be : / ? etc. With
+	the firstc, you know wheither it's a command or a search.
+
```

Comment:
Can't this be deduced from ["cmdline", content, pos] as well ? or given as a parameter of `["cmdline", content, pos,submode={search/reverse-search/...}]`. In fact all the events could be merged into that single one.